hello-pepr-watch/capabilities/watch.e2e.test.ts

```diff
@@ -0,0 +1,126 @@
+import { beforeAll, afterAll, describe, it, jest, expect } from "@jest/globals";
+import { TestRunCfg } from "helpers/src/TestRunCfg";
+import { mins, timed } from "helpers/src/time";
+import { gone } from "helpers/src/resource";
+import { fullCreate, untilTrue } from "helpers/src/general";
+import { K8s, kind } from "kubernetes-fluent-client";
+import { moduleUp, moduleDown, untilLogged, logs } from "helpers/src/pepr";
+const trc = new TestRunCfg(__filename);
+
+/*
+ * The purpose of this test is to demonstrate that the watch module
+ * correctly processes events based on the type of event.
+ */
+
+// createUpdateConfigMap updates the data field of a ConfigMap
+const createUpdateConfigMap = async (name: string) => {
+  await K8s(kind.ConfigMap).Apply({
+    metadata: {
+      name: name,
+      namespace: "pepr-demo",
+    },
+    data: {
+      name: `${name}-updated`,
+    },
+  });
+}
+
+// deleteCM deletes a ConfigMap by a given name and namespace
+const deleteCM = async (name: string, namespace: string) => {
+  await K8s(kind.ConfigMap).InNamespace(namespace).Delete(name)
+}
+
+// untilUpdatedCM checks if a ConfigMap has been updated by checking the data field
+const untilUpdatedCM = async (name: string, namespace: string): Promise<boolean> => {
```

Comment:
```suggestion
const updatedCM = async (name: string, namespace: string): Promise<boolean> => {
```
Rename to better capture that this function _isn't_ waiting for anything -- it's just an async function that finds out whether a property has been set or not (there's no looping / retry).  Instead, the "until" part is done later where untilTrue() is used in the test.

Reply:
check